rbd/rbd_options_test.go

```diff
@@ -0,0 +1,156 @@
+	assert.Error(t, err)
+	i, err = options.GetUint64(rbd.RbdImageOptionStripeCount)
+	assert.NoError(t, err)
+	assert.True(t, i == 1)
+	_, err = options.GetString(rbd.RbdImageOptionStripeCount)
+	assert.Error(t, err)
+
+	err = options.SetUint64(rbd.RbdImageOptionJournalOrder, 1)
+	assert.NoError(t, err)
+	err = options.SetString(rbd.RbdImageOptionJournalOrder, "string not allowed")
+	assert.Error(t, err)
+	i, err = options.GetUint64(rbd.RbdImageOptionJournalOrder)
+	assert.NoError(t, err)
+	assert.True(t, i == 1)
+	_, err = options.GetString(rbd.RbdImageOptionJournalOrder)
+	assert.Error(t, err)
+
+	err = options.SetUint64(rbd.RbdImageOptionJournalSplayWidth, 1)
+	assert.NoError(t, err)
+	err = options.SetString(rbd.RbdImageOptionJournalSplayWidth, "string not allowed")
+	assert.Error(t, err)
+	i, err = options.GetUint64(rbd.RbdImageOptionJournalSplayWidth)
+	assert.NoError(t, err)
+	assert.True(t, i == 1)
+	_, err = options.GetString(rbd.RbdImageOptionJournalSplayWidth)
+	assert.Error(t, err)
+
+	err = options.SetUint64(rbd.RbdImageOptionJournalPool, 1)
+	assert.Error(t, err)
+	err = options.SetString(rbd.RbdImageOptionJournalPool, "journal")
+	assert.NoError(t, err)
+	_, err = options.GetUint64(rbd.RbdImageOptionJournalPool)
+	assert.Error(t, err)
+	s, err = options.GetString(rbd.RbdImageOptionJournalPool)
+	assert.NoError(t, err)
+	assert.True(t, s == "journal")
+
+	err = options.SetUint64(rbd.RbdImageOptionFeaturesSet, 1)
+	assert.NoError(t, err)
+	err = options.SetString(rbd.RbdImageOptionFeaturesSet, "string not allowed")
+	assert.Error(t, err)
+	i, err = options.GetUint64(rbd.RbdImageOptionFeaturesSet)
+	assert.NoError(t, err)
+	assert.True(t, i == 1)
+	_, err = options.GetString(rbd.RbdImageOptionFeaturesSet)
+	assert.Error(t, err)
+
+	err = options.SetUint64(rbd.RbdImageOptionFeaturesClear, 1)
+	assert.NoError(t, err)
+	err = options.SetString(rbd.RbdImageOptionFeaturesClear, "string not allowed")
+	assert.Error(t, err)
+	i, err = options.GetUint64(rbd.RbdImageOptionFeaturesClear)
+	assert.NoError(t, err)
+	assert.True(t, i == 1)
+	_, err = options.GetString(rbd.RbdImageOptionFeaturesClear)
+	assert.Error(t, err)
+
+	err = options.SetUint64(rbd.RbdImageOptionDataPool, 1)
+	assert.Error(t, err)
+	err = options.SetString(rbd.RbdImageOptionDataPool, "data")
+	assert.NoError(t, err)
+	_, err = options.GetUint64(rbd.RbdImageOptionDataPool)
+	assert.Error(t, err)
+	s, err = options.GetString(rbd.RbdImageOptionDataPool)
+	assert.NoError(t, err)
+	assert.True(t, s == "data")
+
+	/* introduced with Ceph Mimic, can not be tested on Luminous
+	err = options.SetUint64(rbd.RbdImageOptionFlatten, 1)
+	assert.NoError(t, err)
+	err = options.SetString(rbd.RbdImageOptionFlatten, "string not allowed")
+	assert.Error(t, err)
+	i, err = options.GetUint64(rbd.RbdImageOptionFlatten)
+	assert.NoError(t, err)
+	assert.True(t, i == 1)
+	_, err = options.GetString(rbd.RbdImageOptionFlatten)
+	assert.Error(t, err)
+	*/
+
+	options.Destroy()
```

Comment:
If options must be cleaned up with calls to the ceph api should we
a) Make it a habit of calling Destroy via defer
b) Add a finalizer (as you did in another recent PR) ?

Reply:
In this case `defer options.Destroy()` is good.
The other finalizer that was added is only needed for some cornercase where destroying/shutdown should not be needed (but without it there is a memory leak).